packages/graphql/src/translate/utils/resolveTree.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ResolveTree } from "graphql-parse-resolve-info";
+import { BaseField } from "../../types";
+import { removeDuplicates } from "../../utils/utils";
+
+/* Finds a field of selection based on field name */
+export const fieldNameExistsInSelection = (selection: Record<string, ResolveTree>) => (fieldName: string) =>
+    Object.values(selection).find((resolveTree) => resolveTree.name === fieldName);
+
+/* Finds a aliased field of selection based on field name */
+export const fieldNameAliasedInSelection = (selection: Record<string, ResolveTree>) => (fieldName: string) =>
+    Object.values(selection).find((resolveTree) => resolveTree.name === fieldName && resolveTree.alias !== fieldName);
+
+export const filterFieldsInSelection =
+    (selection: Record<string, ResolveTree>) =>
+    <T extends BaseField>(fields: T[]) =>
+        fields.filter((field) => Object.values(selection).find((f) => f.name === field.fieldName));
+
+/* Generates a field to be used in creating projections */
+export const generateProjectionField = ({ name }: { name: string }): Record<string, ResolveTree> => {
```

Comment:
**suggestion (if-minor):** Now that we have these utils, it could be useful to unit-test them.
I realize that most of these deal with a `ResolveTree` object, which may be hard to mock. Thoughts?

Reply:
Some of these are really simple and straightforward (which I understand should never be a response to whether unit tests are necessary), but otherwise doable.

Reply:
Tests in https://github.com/neo4j/graphql/pull/842/commits/3b465a2ce89074e05baf82bc955d7e92ed27c547